CMakeSettings.json

```
@@ -4,10 +4,12 @@
       "name": "x64-DebugWithTests",
       "generator": "Ninja",
       "configurationType": "Debug",
-      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "inheritEnvironments": [
+        "msvc_x64_x64"
+      ],
       "buildRoot": "${projectDir}\\out\\build\\${name}",
       "installRoot": "${projectDir}\\out\\install\\${name}",
-      "cmakeCommandArgs": "-DINSTALL_GTEST=OFF -DBUILD_TESTING=ON -DBUILD_CURL_TRANSPORT=ON -DBUILD_STORAGE_SAMPLES=ON",
+      "cmakeCommandArgs": "-DINSTALL_GTEST=OFF -DBUILD_TESTING=ON -DBUILD_TRANSPORT_CURL=ON -DBUILD_STORAGE_SAMPLES=ON",
```

There is some value in using same names for options as the C SDK, where possible:
```suggestion
      "cmakeCommandArgs": "-DINSTALL_GTEST=OFF -DBUILD_TESTING=ON -DTRANSPORT_CURL=ON -DBUILD_STORAGE_SAMPLES=ON",
```

---

not really.
And I would prefer updating the one in C repo.
For Cpp we are following the prefix 'BUILD' for all cmake options that enables components to be compiled out/in.
BUILD_DOCUMENTATION, BUILD_TESTING, BUILD_TRANSPORT_CURL...  etc

---

We shouldn't change the C ones since they already shipped GA and smaller option names are easier to remember.

But we can revisit this options rename later (outside this PR) for C++.

---

https://github.com/Azure/azure-sdk-for-cpp/issues/732